### DISCUSSION TITLE
Connection refused when attempting to update row with trigger function through nextjs api call.

### DISCUSSION THREAD

**Discussion by mikelyndon** [supabase/supabase#7065] [General]
I'm attempting to recreate the trigger function setup locally from the egghead course with nextjs, supabase and stripe.
Until function hooks are supported for local development I figured I'd recreate the setup with standard trigger functions.
So, I have a nextjs api function that gets a stripe customer id and then updates the corresponding row in a profile table. I have a trigger function that's fired when a new entry is created in the profile table.

I've tested the api route with thunder client and it works fine. But for some reason when the api route is called from the trigger function it fails with
```json
{
  message: 'FetchError: request to http://localhost:54321/rest/v1/profile?id=eq.d059c71c-faad-405d-80b8-c77d787b3a9d failed, reason: connect ECONNREFUSED 127.0.0.1:54321',
  details: '',
  hint: '',
  code: 'ECONNREFUSED'
}
```

I've checked the supabaseClient to make sure the headers include the apikey and Authorization token.

I'm probably missing something super obvious here but I'm at a loss. 

This is the nextjs api route.
```javascript
import Stripe from "stripe";
import {
  supabaseClient,
} from "@supabase/supabase-auth-helpers/nextjs";

const handler = async (req, res) => {
  if (req.query.API_ROUTE_SECRET !== process.env.API_ROUTE_SECRET) {
    return res.status(401).send("You are not authorised to call this API");
  }
  const stripe = new Stripe(process.env.STRIPE_SECRET_KEY);

  const customer = await stripe.customers.create({
    email: req.body.record.email,
  });

  let { data: profile, error } = await supabaseClient
    .from("profile")
    .update({
      stripe_customer: customer.id,
    })
    .eq("id", req.body.record.id);

  if (error) console.log("error", error);

  if (profile) console.log("profile", profile);

  res.send({ message: `stripe customer created: ${customer.id}` });
};

export default handler;
```

This is what I've got in my migration file

```sql
CREATE OR REPLACE FUNCTION public.get_stripe_customer()
    RETURNS trigger as $$
DECLARE
  url varchar := 'http://host.docker.internal:3000/api/create-stripe-customer?API_ROUTE_SECRET=9874323lkjf90823lkjsf90823kjlf098l23j';
  payload varchar := '{"record": {"email": ' || '"' || NEW.email || '","id": ' || '"' || NEW.id || '"}}';
  response varchar;
BEGIN
  SELECT STATUS INTO response FROM extensions.http_post(url, payload, 'application/json');
  RETURN NEW;
END;
$$ LANGUAGE plpgsql SECURITY DEFINER;

CREATE TRIGGER add_stripe_id_to_profile
    AFTER INSERT
    ON public.profile
    FOR EACH ROW
    EXECUTE procedure public.get_stripe_customer();
```

EDIT:
A small update. I tried creating another table and adding a similar update function to write to the new table. That works.
```javascript
  let { data: test, error: testError } = await supabaseClient
    .from("httptest")
    .update({ quote: customer.id })
    .eq("id", 1);
```
 So my thinking is that there's something about the permissions or another constraint on the profile table that's stopping it from being updated.
I don't have RLS turned on. Any suggestions on what else to look for?
I've also tried setting the owner and permissions on the profile table with the following:
```sql
ALTER TABLE IF EXISTS public.profile
    OWNER to postgres;
GRANT ALL ON TABLE public.profile TO anon;
GRANT ALL ON TABLE public.profile TO authenticated;
GRANT ALL ON TABLE public.profile TO postgres;
GRANT ALL ON TABLE public.profile TO service_role;
GRANT ALL ON TABLE public.profile TO PUBLIC;
```

**Comment by silentworks** [comment]
Where does this URL point to `http://localhost:54321/`, also can you state where the error is happening, is it inside the trigger function or inside the NextJS api route?

**Comment by paulvonber** [comment]
I have exactly the same issue with Nuxt3, and it's also question where this url "http://localhost:54321/" coming from, in my env I have  set proper supabase url SUPABASE_URL=https://{hereismysupabaseid}.supabase.co
